ms2query/app_helpers.py

```diff
@@ -475,9 +468,113 @@ def cached_library_matching(documents_query: List[SpectrumDocument],
     return found_matches_s2v
 
 
+def get_library_similarities(found_match: pd.DataFrame,
+                             documents_library: List[SpectrumDocument],
+                             library_num: int,
+                             output_folder: str = "downloads") -> np.array:
+    """Returns sim matrix as np.array, index order corresponds to found_match
+
+    Args:
+    ------
+    found_match:
+        Dataframe containing the scores of the library matches
+    documents_library:
+        Library spectra in DocumentsLibrary format
+    library_num:
+        The library number, 0 means the example library, 1 and 2 (subset of)
+        AllPositive library
+    output_folder:
+        Location to download/get similarity matrix and metadata from
+    """
+    # pylint: disable=protected-access
+    sim_matrix = None
+    if library_num == 0:
+        test_sim_matrix_file = os.path.join(
+            os.path.split(os.path.dirname(__file__))[0], "tests",
+            "test_found_matches_similarity_matrix.csv")
+        sim_matrix = pd.read_csv(test_sim_matrix_file, index_col=0)
+    elif library_num in (1, 2):
+        # construct the slice of the similarity matrix in order of matches ind
+        # take 100 as a max value, same as in library_matching
+        match_inds = found_match.iloc[:100].index.to_list()
+        match_inchi14 = [documents_library[ind]._obj.get("inchikey")[:14]
+                         for ind in match_inds]
+        sim_slice_inds = get_sim_matrix_lookup(match_inchi14, output_folder)
+        sim_matrix = subset_sim_matrix(sim_slice_inds, output_folder)
+    else:
+        st.write("Similarity matrix not yet implemented for this library.")
+    return sim_matrix
```

Comment:
```suggestion
        return subset_sim_matrix(sim_slice_inds, output_folder)

    return False
```

Reply:
This way we could make this function independent of streamlit (and hence add a test).
The ``st.write("Similarity matrix not yet implemented for this library.")`` part would then have to be moved to where this function is called.

Reply:
Yes that makes sense indeed